tools/runtime_CLI.py

```diff
@@ -332,16 +332,42 @@ def get_json_key(key):
                                                   json_["header_types"])
                 table.key += [(field_name, match_type, bitwidth)]
 
-    for j_meter in get_json_key("meter_arrays"):
-        meter_array = MeterArray(j_meter["name"], j_meter["id"])
-        if "is_direct" in j_meter and j_meter["is_direct"]:
-            meter_array.is_direct = True
-            meter_array.binding = j_meter["binding"]
-        else:
-            meter_array.is_direct = False
-            meter_array.size = j_meter["size"]
-        meter_array.type_ = MeterType.from_str(j_meter["type"])
-        meter_array.rate_count = j_meter["rate_count"]
+    # If extern_instances exists in input json file,
+    # the json file follows PSA format.
+    # Else the json file follows v1model format.
+    if get_json_key("extern_instances"):
+        # PSA meter
+        for j_meter in get_json_key("extern_instances"):
+            if j_meter["type"] != "Meter":
+                continue
+            meter_array = MeterArray(j_meter["name"], j_meter["id"])
+            attribute_values = j_meter.get("attribute_values", [])
+            for attr in attribute_values:
+                name = attr.get("name", [])
+                val_type = attr.get("type", [])
+                value = attr.get("value", [])
+                if name == "is_direct":
+                    meter_array.is_direct = value == True
+                # TODO set meter_array.binding for direct_meter
+                # direct_meter not supported on PSA yet
+                elif name == "n_meters":
+                    meter_array.size = value
+                elif name == "type":
+                    meter_array.type_ = value
+                elif name == "rate_count":
+                    meter_array.rate_count = value
```

Comment:
architecture-specific extern stuff belongs in the CLI extension file. For PSA it is: https://github.com/p4lang/behavioral-model/blob/master/targets/psa_switch/pswitch_CLI.py

I suggest adding this logic after this line: https://github.com/p4lang/behavioral-model/blob/master/targets/psa_switch/pswitch_CLI.py#L91 and instantiating `MeterArray` there as appropriate

Reply:
This makes a lot of sense, I was thinking that PSA and v1model shouldn't just share `tools/runtime_CLI.py`, thanks

Reply:
@antoninbas

All fixed:  
* Moved psa-meter json parsing to `pswitch_CLI.py`
* Minor updates on coding style

Would you please take a look and let me know if any further change is needed? Thanks!